kvdb/etcd/stm.go

```diff
@@ -262,8 +282,11 @@ func runSTM(s *stm, apply func(STM) error) error {
 				return
 			}
 
-			// Rollback before trying to re-apply.
-			s.Rollback()
+			// Reset the write set before trying to re-apply. Upon
+			// commit we retrieved the latest version of all
+			// previously fetched keys and ranges so we don't need
+			// to reset the read set.
+			s.reset(false)
```

Comment:
Any reason to go w/ the reset naming vs roll back? Latter used elsewhere in the other DB backends, keeps the same metaphorical semantics.